packages/core/src/searchWithRetries.ts

```diff
@@ -17,6 +17,12 @@ export async function searchWithRetries(api: API, query: SearchQuery, options: O
         throw error
       }
 
+      if (retries >= maxRetries && options.fallback) {
```

Comment:
not sure if we need fallback logic at this level
after maxRetries the error will be throw and fallbacks can be applied at higher level